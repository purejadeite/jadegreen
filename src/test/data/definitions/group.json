{
	"id": "book",
	"sheets": [
		{
			"comment": "グループ",
			"id": "sub",
		    "name": "validation",
		    "cells": [
		        {"id": "logicalClassName", "row": 1, "column": 3},
		        {"id": "className", "row": 2, "column": 3},
		        {"id": "fields", "beginRow": 6, "endKeyId": "fieldName", "options": [{"type": "group", "keyId": "fieldName", "toId": "validators"}],
		            "columns": [
		                {"id": "logicalFieldName", "column": 2},
		                {"id": "fieldName", "column": 3},
		                {"id": "validationType", "column": 4},
		                {"id": "arg0", "column": 5},
		                {"id": "arg1", "column": 6},
		                {"id": "arg2", "column": 7},
		                {"id": "arg3", "column": 8}
		            ]
		        }
		    ]
		}
	]
}