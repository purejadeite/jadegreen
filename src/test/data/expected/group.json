[
	{
		"sheetName": "validation",
		"logicalClassName": "グループ化",
		"className": "Sample",
		"fields": [
			{
				"fieldName": "filePath",
				"validators": [
					{
						"logicalFieldName": "ファイルパス",
						"fieldName": "filePath",
						"validationType": "notnull",
						"arg0": null,
						"arg1": null,
						"arg2": null,
						"arg3": null
					},
		   			{
						"logicalFieldName": "ファイルパス",
						"fieldName": "filePath",
						"validationType": "length",
						"arg0": "2",
						"arg1": "10",
						"arg2": null,
						"arg3": null
					}
				]
			},
			{
				"fieldName": "fileName",
				"validators": [
		   			{
		   				"logicalFieldName": "ファイル名",
						"fieldName": "fileName",
						"validationType": "notnull",
						"arg0": null,
						"arg1": null,
						"arg2": null,
						"arg3": null
					}
				]
			},
			{
				"fieldName": "fileExistFlag",
				"validators": [
		   			{
						"logicalFieldName": "ファイル有無フラグ",
						"fieldName": "fileExistFlag",
						"validationType": "notnull",
						"arg0": null,
						"arg1": null,
						"arg2": null,
						"arg3": null
					}
				]
			},
			{
				"fieldName": "lines",
				"validators": [
		   			{
						"logicalFieldName": "行情報",
						"fieldName": "lines",
						"validationType": "notnull",
						"arg0": null,
						"arg1": null,
						"arg2": null,
						"arg3": null
					},
					{
						"logicalFieldName": "行情報",
						"fieldName": "lines",
						"validationType": "maxlength",
						"arg0": "100",
						"arg1": null,
						"arg2": null,
						"arg3": null
					},
					{
						"logicalFieldName": "行情報",
						"fieldName": "lines",
						"validationType": "kana",
						"arg0": null,
						"arg1": null,
						"arg2": null,
						"arg3": null
					}
				]
			}
		]
	}
]